tests/migrations/test_executor.py

```diff
@@ -368,6 +368,44 @@ def callback(*args):
         ]
         self.assertEqual(call_args_list, expected)
 
+    @override_settings(
+        INSTALLED_APPS=[
+            "migrations.migrations_test_apps.alter_fk.author_app",
+            "migrations.migrations_test_apps.alter_fk.book_app",
+        ]
+    )
+    def test_alter_id_type_with_fk(self):
+        try:
+            executor = MigrationExecutor(connection)
+            self.assertTableNotExists("author_app_author")
+            self.assertTableNotExists("book_app_book")
+            # Apply initial migrations
+            executor.migrate([
+                ("author_app", "0001_initial"),
+                ("book_app", "0001_initial"),
+            ])
+            self.assertTableExists("author_app_author")
+            self.assertTableExists("book_app_book")
+            # Rebuild the graph to reflect the new DB state
+            executor.loader.build_graph()
+
+            # Apply PK type alteration
+            executor.migrate([("author_app", "0002_alter_id")])
+
+            # Rebuild the graph to reflect the new DB state
+            executor.loader.build_graph()
+        finally:
+            pass
+            # Cleanup
+            # TODO: PostgreSQL can't reverse the AlterField operation because
```

Comment:
You mean manually? Should be possible I guess.

On February 12, 2015 4:05:00 PM GMT+01:00, Tim Graham <notifications@github.com> wrote:
>> +                ("book_app", "0001_initial"),
>> +            ])
>> +            self.assertTableExists("author_app_author")
>> +            self.assertTableExists("book_app_book")
>> +            # Rebuild the graph to reflect the new DB state
>> +            executor.loader.build_graph()
>> +
>> +            # Apply PK type alteration
>> +            executor.migrate([("author_app", "0002_alter_id")])
>> +
>> +            # Rebuild the graph to reflect the new DB state
>> +            executor.loader.build_graph()
>> +        finally:
>> +            pass
>> +            # Cleanup
>> +            # TODO: PostgreSQL can't reverse the AlterField
>operation because
>
>Could you fake reverse 0002 and reverse to zero to cleanup the tables?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/django/django/pull/4097/files#r24586582